platform/security/src/test/java/org/codice/alliance/security/claims/videographer/TestVideographerClaimsHandler.java

```diff
@@ -0,0 +1,146 @@
+
+    private static final String CLAIM_URI_3 =
+            "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname";
+
+    private static final String CLAIM_VALUE_1 = "Videographer";
+
+    private static final String CLAIM_VALUE_2a = "Videographer@videographer.com";
+
+    private static final String CLAIM_VALUE_2b = "someguy@somesite.com";
+
+    private static final String CLAIM_VALUE_2c = "somedude@cool.com";
+
+    private static final String CLAIM_VALUE_3 = "Videographer";
+
+    private static final String CLAIM1 = CLAIM_URI_1 + "=" + CLAIM_VALUE_1;
+
+    private static final String CLAIM2 = CLAIM_URI_2 + "=" + CLAIM_VALUE_2a;
+
+    private static final String CLAIM3 = CLAIM_URI_3 + "=" + CLAIM_VALUE_3;
+
+    @Test
+    public void testSettingClaimsMapList() throws URISyntaxException {
+        VideographerClaimsHandler claimsHandler = new VideographerClaimsHandler();
+        claimsHandler.setAttributes(Arrays.asList(CLAIM1, CLAIM2, CLAIM3));
+
+        Map<URI, List<String>> claimsMap = claimsHandler.getClaimsMap();
+
+        List<String> value = claimsMap.get(new URI(CLAIM_URI_1));
+        assertThat(value.get(0), is(CLAIM_VALUE_1));
+
+        value = claimsMap.get(new URI(CLAIM_URI_2));
+        assertThat(value.get(0), is(CLAIM_VALUE_2a));
+
+        value = claimsMap.get(new URI(CLAIM_URI_3));
+        assertThat(value.get(0), is(CLAIM_VALUE_3));
+
+    }
+
+    @Test
+    public void testRetrieveClaims() throws URISyntaxException {
+        VideographerClaimsHandler claimsHandler = new VideographerClaimsHandler();
+        claimsHandler.setAttributes(Arrays.asList(CLAIM1,
+                CLAIM_URI_2 + "=" + CLAIM_VALUE_2a + "|" + CLAIM_VALUE_2b + "|" + CLAIM_VALUE_2c,
+                CLAIM3));
+
+        ClaimCollection requestClaims = new ClaimCollection();
+        Claim requestClaim = new Claim();
+        URI nameURI = new URI(CLAIM_URI_1);
+        requestClaim.setClaimType(nameURI);
+        requestClaims.add(requestClaim);
+        requestClaim = new Claim();
+        URI emailURI = new URI(CLAIM_URI_2);
+        requestClaim.setClaimType(emailURI);
+        requestClaims.add(requestClaim);
+        requestClaim = new Claim();
+        URI fooURI = new URI("http://schemas.xmlsoap.org/ws/2005/05/identity/claims/foobar");
+        requestClaim.setClaimType(fooURI);
+        requestClaim.setOptional(true);
+        requestClaims.add(requestClaim);
+
+        ClaimsParameters claimsParameters = new ClaimsParameters();
+        claimsParameters.setPrincipal(new VideographerPrincipal("127.0.0.1"));
+
+        List<URI> supportedClaims = claimsHandler.getSupportedClaimTypes();
+
+        assertThat(supportedClaims, hasSize(3));
+
+        ProcessedClaimCollection claimsCollection = claimsHandler.retrieveClaimValues(requestClaims,
+                claimsParameters);
+
+        assertThat(claimsCollection, hasSize(3));
+
+        for (ProcessedClaim claim : claimsCollection) {
```

Comment:
This test won't fail if your claim collection doesn't contain the expected claims. I'd recommend something like this:
```
Matcher<ProcessedClaim> nameClaim = new CustomMatcher<ProcessedClaim>("claim by name (or some better description)") {
    @Override
    public boolean matches(Object o) {
        ProcessedClaim claim = (ProcessedClaim) o;
        // return true if all the expectations are met
    }
};
... write your other matchers for the claims you expect
assertThat(claimsCollection, containsInAnyOrder(nameClaim, emailClaim, ...));
```